snuba/query/processors/tagsmap.py

```diff
@@ -0,0 +1,125 @@
+import re
+
+from enum import Enum
+from typing import Optional, List, NamedTuple
+
+from snuba.query.query import Query
+from snuba.query.query_processor import QueryProcessor
+from snuba.datasets.transactions_processor import escape_field
+from snuba.query.types import Condition
+from snuba.request.request_settings import RequestSettings
+from snuba.util import is_condition, is_function
+
+
+TAG_PATTERN = re.compile(r"([a-zA-Z0-9_\.]+)\[([a-zA-Z0-9_\.]+)\]")
+
+
+class Operand(Enum):
+    EQ = "="
+    NEQ = "!="
+
+
+class OptimizableCondition(NamedTuple):
+    nested_col_index: str
+    operand: Operand
+    value: str
+
+
+class NestedFieldConditionOptimizer(QueryProcessor):
+    """
+    This processor scans the the conditions in the query and converts
+    top level eq/neq conditions on specific nested fields, into LIKE/NOT LIKE
+    statements on a column that merges all the content of a nested one.
+
+    Example:
+    tags: {
+        "tag1": "1",
+        "tag2": "2",
+    }
+
+    The merged columns looks like:
+    tags_map: "tag1:1|tag2:2"
+
+    This means that a simple EQ/NEQ condition like `tags[tag1] == 1` can be
+    transformed into `tags_map LIKE "%tag1:2%"`
+    """
+
+    def __init__(self, nested_col: str, merged_col: str) -> None:
+        self.__nested_col = nested_col
+        self.__merged_col = merged_col
+
+    def __is_optimizable(
+        self, condition: Condition, column: str
+    ) -> Optional[OptimizableCondition]:
+        """
+        Recognize if the condition can be optimized.
+        This includes these kind of conditions:
+        - top level conditions. No nested OR
+        - the condition has to be either in the form tag[t] = value
+        - functions referencing the tags as parameters are not taken
+          into account except for ifNull.
+        - Both EQ and NEQ conditions are optimized.
+        """
+        if not is_condition(condition):
+            return None
+        if condition[1] not in ["=", "!="]:
```

Comment:
```suggestion
        if condition[1] not in [Operand.EQ.value, Operand.NEQ.value]:
```

Or is there a reason these operators are not assumed to be reflexive? I noticed this on like 88 below, so seems like it may have been intentional?

Reply:
No reason, I just introduced the enum later and forgot this spot.